### PR TITLE
feat: add delegate in request client

### PR DESCRIPTION
**Pull request by vrolland**

### CODE REVIEW

File: packages/integration-test/test/node-client.test.ts [hunk 1]
```diff
@@ -611,3 +630,104 @@ describe('ERC20 localhost request creation and detection test', () => {
...
+    currency: {
+      type: Types.RequestLogic.CURRENCY.ISO4217,
+      value: 'EUR',
+    },
+    expectedAmount: '1000',
+    payee: payeeIdentity,
+    payer: payerIdentity,
+  };
+
+  it('can create an declarative request and declare payment and refund from payer & payee', async () => {
```
Comment (yomarion) [inline packages/integration-test/test/node-client.test.ts:652]:
```suggestion
  it('can create a declarative request and declare payment and refund from payer & payee', async () => {
```

File: packages/integration-test/test/node-client.test.ts [hunk 2]
```diff
@@ -611,3 +630,104 @@ describe('ERC20 localhost request creation and detection test', () => {
...
+    requestData = await request.declareReceivedPayment('1000', 'transfer received', payeeIdentity);
+    requestData = await new Promise((resolve): any => requestData.on('confirmed', resolve));
+    expect(requestData.balance!.balance).toBe('1000');
+
+    requestData = await request.declareReceivedRefund('900', 'refund received', payerIdentity);
+    requestData = await new Promise((resolve): any => requestData.on('confirmed', resolve));
+    expect(requestData.balance!.balance).toBe('100');
+  });
+
+  it('can create an declarative request and declare payment delegatePayee', async () => {
```
Comment (yomarion) [inline packages/integration-test/test/node-client.test.ts:686]:
```suggestion
  it('can create a declarative request and declare payment by delegatePayee', async () => {
```

File: packages/integration-test/test/node-client.test.ts [hunk 3]
```diff
@@ -611,3 +630,104 @@ describe('ERC20 localhost request creation and detection test', () => {
...
+    requestData = await request.declareReceivedPayment(
+      '1000',
+      'transfer received',
+      delegatePayeeIdentity,
+    );
+    requestData = await new Promise((resolve): any => requestData.on('confirmed', resolve));
+    expect(requestData.balance!.balance).toBe('1000');
+  });
+
+  it('can create an declarative request and declare refund delegatePayer', async () => {
```
Comment (yomarion) [inline packages/integration-test/test/node-client.test.ts:710]:
```suggestion
  it('can create a declarative request and declare refund by delegatePayer', async () => {
```

**Review by yomarion — APPROVED** [review]
Nice one !

File: packages/request-client.js/src/api/request.ts [hunk 4]
```diff
@@ -656,6 +656,65 @@ export default class Request {
     return requestData;
   }
 
+  /**
+   * Declare a refund is received for the declarative payment network
```
Comment (benjlevesque) [inline packages/request-client.js/src/api/request.ts:660]: wrong comment? It's important to clearly describe this function, as its name could be confusing. 

Maybe rename to `addDeclarativeDelegate`?

**Review by benjlevesque — APPROVED** [review]

**Pull request merged**